elastic/apm-server
# Add X-Elastic-Product-Origin header to es requests

## Motivation/summary
Add the `X-Elastic-Product-Origin` header to requests made to Elasticsearch to identify them as originating from an Elastic product and ensure any future Elasticsearch API deprecation warnings that may occur from internal use are not presented to users.

## How to test these changes

1. start elasticsearch
2. start the following proxy:
    ```golang
	package main

	import (
		"fmt"
		"log"
		"net/http"
		"net/http/httputil"
		"net/url"
	)

	func main() {
		u, err := url.Parse("http://0.0.0.0:9200")
		if err != nil {
			panic(err)
		}
		proxy := new(httputil.ReverseProxy)
		proxy.Director = func(r *http.Request) {
			req, err := httputil.DumpRequest(r, false)
			if err != nil {
				panic(err)
			}
			fmt.Printf(string(req))
			r.URL = u
		}

		log.Fatal(http.ListenAndServe(":9201", proxy))
	}
    ```
3. start apm-server configured to have an elasticsearch output at `http://localhost:9200`
4. send some events
5. check the request written to console by the proxy, verify the `X-Elastic-Product-Origin` header is present

## Related issues

closes #7131
